### DISCUSSION TITLE
MITM proxy and capturing requests (HTTPS) - certificate problem

### DISCUSSION THREAD

**Discussion by K3v1n369** [mitmproxy/mitmproxy#7169] [General]
Hello,
I installed MITM proxy, because I wanted to know, if my browser Mullvad (Firefox fork) sends telemetry requests even if telemetry is disabled. After installation of MITM proxy, certificate was generated, I imported it to the browser. Browser is configured for MITM proxy usage.

When I tried to put different web address (with HTTPS protocol) to the browser, so I got the following warning:

**Warn: [time] [IP_address:port] client tls handshake failed. the client does not trust the proxy's certificate for "domain_name" (ssl/tls alert bad certificate)**

OR

**warn:  [time] [IP_address:port] client tls handshake failed. the client does not trust the proxy's certificate for "domain_name" (tlsv1 alert unknown ca)**

I want to add, that even I can see these warning in MITM console, so the web pages are loaded and the console shows the HTTPS requests.

My questions are:
1) Will I see all HTTPS requests in console, even if the warnings are displayed? Can I ignore the warnings?
2) How to fix this problem with certificate?

Thank you very much
![handshake1](https://github.com/user-attachments/assets/9d8fc27c-b41a-4e3f-9df0-f8e9cb7c08aa)
![handshake](https://github.com/user-attachments/assets/d609e6be-742c-4cb5-8b6c-4cd83bcdaa00)

**Comment by eimanes** [comment]
it also happens for IOS devices

**Reply by K3v1n369** [reply]
Solved, steps below:

1) In Mullvad browser put to address line: **about:config**
2) Find **security.nocertdb** and set value to: **false**
3) Import certificate
4) Restart browser

Source: [https://github.com/mullvad/mullvad-browser/issues/76#issuecomment-1589487320-permalink](url)
